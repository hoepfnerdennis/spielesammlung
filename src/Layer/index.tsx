import React, { useEffect, useRef } from 'react';
import ReactDOM from 'react-dom';
import styles from './styles.module.css';

const Layer: React.SFC<{ children: React.ReactNode; closeLayer: () => void }> = ({
  children,
  closeLayer,
}): JSX.Element => {
  const root = useRef<HTMLElement | null>(document.getElementById('layer'));
  const container = useRef<HTMLDivElement>(document.createElement('div'));

  useEffect(() => {
    if (root.current) {
      const currentRoot = root.current;
      const currentContainter = container.current;
      currentRoot.appendChild(currentContainter);
      return (): void => {
        currentRoot.removeChild(currentContainter);
      };
    }
    return (): void => {
      // do nothing
    };
  }, []);

  return ReactDOM.createPortal(
    <div className={styles.layer}>
      <button type="button" className={styles.background} onClick={closeLayer} />
      <div className={styles.container}>{children}</div>
    </div>,
    container.current
  );
};

export default Layer;
