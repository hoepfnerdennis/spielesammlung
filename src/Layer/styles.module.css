.layer {
    position: fixed;
    top: 0px;
    bottom: 0px;
    left: 0px;
    right: 0px;
    background-color: rgba(0, 139, 139, 0.5);
    z-index: 1000;
    width: 100%;
    height: 100%;
    border: none;

    display: flex;
    justify-content: center;
    align-items: center;
}

.background {
    position: absolute;
    background: none;
    border: none;
    width: 100%;
    height: 100%;
}

.container {
    background-color: white;
    max-width: 300px;
    width: 100%;

    /* height: 100%; */
    max-height: 50vh;

    overflow-y: scroll;

    padding: 1rem;
    z-index: 1001;
}