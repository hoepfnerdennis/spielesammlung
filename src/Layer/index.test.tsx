import React from 'react';
import { render } from '@testing-library/react';
import Layer from './index';

jest.mock('react-dom', () => ({
  ...jest.requireActual('react-dom'),
  // eslint-disable-next-line @typescript-eslint/explicit-function-return-type,@typescript-eslint/no-explicit-any
  createPortal: (children: any) => <div data-testid="portal">{children}</div>,
}));

describe('Layer', () => {
  it('should render and match snapshot', () => {
    const layerRoot = document.createElement('div');
    layerRoot.setAttribute('id', 'layer');
    const { container } = render(<Layer closeLayer={jest.fn()}>children</Layer>, {
      container: layerRoot,
    });
    expect(container).toMatchSnapshot();
  });
});
