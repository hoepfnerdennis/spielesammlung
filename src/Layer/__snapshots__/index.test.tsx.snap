// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layer should render and match snapshot 1`] = `
<div
  id="layer"
>
  <div
    data-testid="portal"
  >
    <div
      class="layer"
    >
      <button
        aria-label="layer schlieÃŸen"
        class="background"
        type="button"
      />
      <div
        class="container"
      >
        children
      </div>
    </div>
  </div>
</div>
`;
